src/test/java/uk/gov/pay/ledger/queue/QueueMessageReceiverIT.java

```diff
@@ -84,4 +84,34 @@ public void shouldContinueToHandleMessagesFromQueueForDownstreamExceptions() thr
                 .body("charge_id", is(resourceExternalId2))
                 .body("state.status", is("created"));
     }
+
+    @Test
+    public void shouldHandleRefundEvent() throws InterruptedException {
+        final String resourceExternalId = "rexid";
+        final String parentResourceExternalId = "parentRexId";
+        aQueueEventFixture()
+                .withResourceExternalId(parentResourceExternalId)
+                .withEventDate(CREATED_AT)
+                .withEventType("AUTHORISATION_SUCCESSFUL")
+                .insert(rule.getSqsClient());
+
+        Thread.sleep(100);
+        aQueueEventFixture()
+                .withResourceExternalId(resourceExternalId)
+                .withParentResourceExternalId(parentResourceExternalId)
+                .withEventDate(CREATED_AT)
+                .withEventType("REFUND_CREATED_BY_SERVICE")
+                .withResourceType(ResourceType.REFUND)
+                .insert(rule.getSqsClient());
+
+        Thread.sleep(500);
+
+        given().port(rule.getAppRule().getLocalPort())
+                .contentType(JSON)
+                .get("/v1/transaction/" + resourceExternalId)
+                .then()
+                .statusCode(200)
+                .body("charge_id", is(resourceExternalId))
```

Comment:
Why does the refund have a `charge_id`? If this isn't expected behaviour potentially there should be a comment that reminds us to update this once the read end points have been updated.

Reply:
We don't have separate endpoint to search for refunds. Added a comment to use refunds endpoint when available.

Reply:
Great - I'm doing to add a ticket just to make sure!